base_dj/models/base.py

```diff
@@ -197,27 +199,32 @@ def is_missing(r):
                 (record, to_xid(record.id))
                 for record in self
             )
-
         xids.update(
             (r.id, (modname, r._dj_xmlid_export_name())) for r in missing
         )
         # you can generate one shot xids and not store them
         # so you don't pollute your db and maybe fix some csv
         if not self.env.context.get('dj_xmlid_skip_create'):
             fields = ['module', 'model', 'name', 'res_id']
-            self.env.cr.copy_from(io.StringIO(
-                u'\n'.join(
-                    u"%s\t%s\t%s\t%d" % (
-                        modname,
-                        record._name,
-                        xids[record.id][1],
-                        record.id,
-                    )
-                    for record in missing
-                )),
-                table='ir_model_data',
-                columns=fields,
-            )
+            try:
+                self.env.cr.copy_from(io.StringIO(
+                    u'\n'.join(
+                        u"%s\t%s\t%s\t%d" % (
+                            modname,
+                            record._name,
+                            xids[record.id][1],
+                            record.id,
+                        )
+                        for record in missing
+                    )),
+                    table='ir_model_data',
+                    columns=fields,
+                )
+            except IntegrityError:
+                raise UserError(
+                    "Writing xmlids for %s failed."
+                    " Probably your xmlids aren't unique" % missing[0]._name
```

Comment:
I'm missing context but as I understand it, `cr.copy_from` will trigger an `INSERT INTO ir_model_data ...` statement for each record in `missing`.
We're then not sure which record or SQL statement triggers the error, so are we really sure `_name` is the same for each of these :question: 
If that's always the case it's fine, otherwise you probably should use `mapped` ...

Reply:
premise: this is std code from odoo. It's used to make the bulk insert more efficient.
now, here we are in a single model so what you'll get in "missing" will be just records of `self._name` -> I'd use this for the sake of clarity.